layouts/community/ortho_4x4/jotix/keymap.c

```diff
@@ -1,41 +1,82 @@
 #include QMK_KEYBOARD_H
 
-extern keymap_config_t keymap_config;
+enum layers {
+  _NUMPAD,
+  _GAMEPAD,
+  _FN
+};
 
-// Each layer gets a name for readability, which is then used in the keymap matrix below.
-// The underscores don't mean anything - you can have a layer called STUFF or any other name.
-// Layer names don't all need to be of the same length, obviously, and you can also skip them
-// entirely and just use numbers.
+static bool is_p0_pressed;
+static bool is_p7_pressed;
+static bool is_pmns_pressed;
 
-#define _NUMPAD 0
-#define _LOWER  1
-#define LOWER  LT(_LOWER, KC_PENT)
+#define TGGAME TG(_GAMEPAD)
 
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 
-/* Numpad
- * +-------+-------+-------+-------+
- * |   7   |   8   |   9   |   -   |
- * +-------+-------+-------+-------+
- * |   4   |   5   |   6   |   +   |
- * +-------+-------+-------+-------+
- * |   1   |   2   |   3   |   /   |
- * +-------+-------+-------+-------+
- * |       |   0   |   .   |   *   |
- * +-------+-------+-------+-------+
- */
 [_NUMPAD] = LAYOUT_ortho_4x4 ( 
-    KC_P7,  KC_P8,  KC_P9,  KC_PMNS,
-    KC_P4,  KC_P5,  KC_P6,  KC_PPLS,
-    KC_P1,  KC_P2,  KC_P3,  KC_PSLS,
-    LOWER,  KC_P0,  KC_PDOT,KC_PAST 
+  KC_P7,  KC_P8,  KC_P9,  KC_PMNS,
+  KC_P4,  KC_P5,  KC_P6,  KC_PPLS,
+  KC_P1,  KC_P2,  KC_P3,  MO(_FN),
+  KC_P0,  KC_PDOT,KC_PSLS,KC_PAST 
 ),
 
-/* Lower */
-[_LOWER] = LAYOUT_ortho_4x4 ( 
-    KC_NLCK,_______,_______,RESET  ,
-    _______,_______,_______,_______,
-    _______,_______,_______,_______,
-    _______,_______,_______,_______ 
+[_GAMEPAD] = LAYOUT_ortho_4x4 (
+  KC_5,   KC_6,   KC_7,   KC_8,
+  KC_1,   KC_2,   KC_3,   KC_4,
+  KC_Q,   KC_W,   KC_E,   _______,
+  KC_A,   KC_S,   KC_D,   KC_SPC 
 ),
+
+[_FN] = LAYOUT_ortho_4x4 (
+  KC_PERC,KC_CIRC,KC_AMPR,KC_ASTR,
+  KC_EXLM,KC_AT,  KC_HASH,KC_DLR,
+  KC_Z,   KC_X,   KC_C,   _______,
+  KC_NLCK,_______,_______,TGGAME
+)
 };
+
+uint32_t layer_state_set_user(uint32_t state) {
+  #ifdef JOTPAD16_LEDS
+  if (biton32(state) == _FN) {
+    writePinHigh(JOTPAD16_LED1);
+  } else {
+    writePinLow(JOTPAD16_LED1);
+  }
+  if (biton32(state) == _GAMEPAD) {
+    writePinHigh(JOTPAD16_LED2);
+  } else {
+    writePinLow(JOTPAD16_LED2);
+  }
+  #endif
+  return state;
+}
```

Comment:
Actually, this entire thing could be written as:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
#ifdef JOTPAD16_LEDS
    writePin(JOTPAD16_LED1, (get_highest_layer(state) == _FN));
    writePin(JOTPAD16_LED2, (get_highest_layer(state) == _GAMEPAD));
#endif
  return state;
}
```